src/lsm/tree_fuzz.zig

```diff
@@ -0,0 +1,432 @@
+                    _ = model.remove(key_from_value(&value));
+                },
+                .get => |key| {
+                    // Get account from lsm.
+                    const tree_value = env.get(key);
+
+                    // Compare result to model.
+                    const model_value = model.get(key);
+                    if (model_value == null) {
+                        assert(tree_value == null);
+                    } else {
+                        assert(std.mem.eql(
+                            u8,
+                            std.mem.asBytes(&model_value.?),
+                            std.mem.asBytes(tree_value.?),
+                        ));
+                    }
+                },
+            }
+        }
+    }
+};
+
+pub fn run_fuzz_ops(fuzz_ops: []const FuzzOp) !void {
+    // Init mocked storage.
+    var storage = try Storage.init(
+        allocator,
+        Environment.size_max,
+        Storage.Options{
+            // We don't apply storage faults yet, so this seed doesn't matter.
+            .seed = 0xdeadbeef,
+            .read_latency_min = 0,
+            .read_latency_mean = 0,
+            .write_latency_min = 0,
+            .write_latency_mean = 0,
+            .read_fault_probability = 0,
+            .write_fault_probability = 0,
+        },
+        0,
+        .{
+            .first_offset = 0,
+            .period = 0,
+        },
+    );
+    defer storage.deinit(allocator);
+
+    try Environment.format(&storage);
+    try Environment.run(&storage, fuzz_ops);
+}
+
+fn random_id(random: std.rand.Random, comptime Int: type) Int {
+    // We have two opposing desires for random ids:
+    const avg_int: Int = if (random.boolean())
+        // 1. We want to cause many collisions.
+        8
+    else
+        // 2. We want to generate enough ids that the cache can't hold them all.
+        Environment.cache_entries_max;
+    return fuzz.random_int_exponential(random, Int, avg_int);
+}
+
+pub fn generate_fuzz_ops(random: std.rand.Random) ![]const FuzzOp {
+    const fuzz_op_count = @minimum(
+        @as(usize, 1E7),
+        fuzz.random_int_exponential(random, usize, 1E6),
+    );
+    const fuzz_ops = try allocator.alloc(FuzzOp, fuzz_op_count);
+    errdefer allocator.free(fuzz_ops);
+
+    const fuzz_op_distribution = fuzz.random_enum_distribution(random, std.meta.Tag(FuzzOp));
+    log.info("fuzz_op_distribution = {d:.2}", .{fuzz_op_distribution});
+
+    // We're not allowed to go more than Environment.cache_entries_max puts without compacting.
+    var puts_since_compact: usize = 0;
+
+    for (fuzz_ops) |*fuzz_op| {
+        fuzz_op.* = if (puts_since_compact >= Environment.cache_entries_max)
+            // We have to compact before doing any other operations.
+            FuzzOp{ .compact = {} }
+        else
+        // Otherwise pick a random FuzzOp.
+        switch (fuzz.random_enum(random, std.meta.Tag(FuzzOp), fuzz_op_distribution)) {
```

Comment:
I'm not sure of the best way to do this, but I think we will find more interesting errors if we vary the tree workload (as generated by the FuzzOps) in other interesting ways. Some examples:

- Insert a bunch of keys (multiple commits worth), then remove all/most of those keys. (Right now this would crash compaction, though that will be fixed by King's iterator PR).
- For key selection, it isn't merged yet, but the VOPR-workload tests [various interesting id patterns](https://github.com/tigerbeetledb/tigerbeetle/blob/dj-vopr-workload/src/test/id.zig) (uniform random, zigzig, ascending, descending). 
  - (Though the VOPR-workload has some extra requirements that this test doesn't; it hides some data in the id to use later for verification.)
- One common pattern in our secondary indexes: composite keys that are the tuple `(small-cardinality value, strictly increasing timestamp)` (This pattern applies to e.g.: `Account.ledger`, `Account.code`).
- Probably others I haven't thought of!